src/components/modals/login-modal/LoginModal.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import { Modal } from 'antd';
+import './login-modal.less';
+import LoginContent from './LoginContent';
+import ForgotPasswordContent from './ForgotPasswordContent';
+import PasswordResetContent from './PasswordResetContent';
+
+interface LoginModalProps {
+  isShowing: boolean;
+  hide: () => void;
+}
+
+const renderContent = (
+  currentPage: string,
+  switchToForgotPasswordPage: () => void,
+  switchToPasswordResetPage: () => void,
+) => {
+  switch (currentPage) {
+    case 'login':
+      return (
+        <LoginContent switchToForgotPasswordPage={switchToForgotPasswordPage} />
+      );
+    case 'forgotPassword':
+      return (
+        <ForgotPasswordContent
+          switchToConfirmationPage={switchToPasswordResetPage}
+        />
+      );
+    case 'passwordReset':
+      return <PasswordResetContent />;
+  }
+};
+
+const LoginModal = (props: LoginModalProps) => {
+  const [currentPage, setPage] = useState('login');
```

Comment:
I think we might just be able to do this with `useLocation`. 
```suggestion
const location = useLocation();

const modalContent: LoginContent | ForgotPasswordContent | PasswordResetContent = getModalContent(location.pathname)
```

Then instead of having the setter functions we can just redirect the user to a new route, which will propagate changes here